src/MongoDB/ClientEncryption.c

```diff
@@ -106,6 +237,223 @@ static PHP_METHOD(MongoDB_Driver_ClientEncryption, decrypt)
+	if (!phongo_query_init(&query, NULL, NULL)) {
+		/* Exception already thrown */
+		goto cleanup;
+	}
+
+	cursor = mongoc_client_encryption_get_keys(intern->client_encryption, &error);
+
+	if (!cursor) {
+		phongo_throw_exception_from_bson_error_t(&error);
+		goto cleanup;
+	}
+
+	if (!phongo_cursor_init_for_query(return_value, &intern->key_vault_client_manager, cursor, intern->key_vault_namespace, &query, NULL, NULL)) {
+		/* Exception already thrown */
+		mongoc_cursor_destroy(cursor);
+		goto cleanup;
+	}
+
+cleanup:
+	zval_ptr_dtor(&query);
+} /* }}} */
+
+/* {{{ proto object|null MongoDB\Driver\ClientEncryption::removeKeyAltName(MongoDB\BSON\Binary $id, string $keyAltName)
+   Removes a keyAltName from the keyAltNames array of the key document in the
+   key vault collection with the given UUID (BSON binary subtype 0x04). Returns
+   the previous version of the key document, or null if no document matched. */
+static PHP_METHOD(MongoDB_Driver_ClientEncryption, removeKeyAltName)
+{
+	zval*        zkeyid         = NULL;
+	char*        keyaltname     = NULL;
+	size_t       keyaltname_len = 0;
+	bson_value_t keyid          = { 0 };
+	bson_t       key_doc        = BSON_INITIALIZER;
+	bson_error_t error          = { 0 };
+
+	PHONGO_PARSE_PARAMETERS_START(2, 2)
+	Z_PARAM_OBJECT_OF_CLASS(zkeyid, php_phongo_binary_ce)
+	Z_PARAM_STRING(keyaltname, keyaltname_len);
+	PHONGO_PARSE_PARAMETERS_END();
+
+	php_phongo_zval_to_bson_value(zkeyid, PHONGO_BSON_NONE, &keyid);
+
+	if (EG(exception)) {
+		goto cleanup;
+	}
+
+	if (!validate_keyid(&keyid)) {
+		/* Exception already thrown */
+		goto cleanup;
+	}
+
+	if (!mongoc_client_encryption_remove_key_alt_name(Z_CLIENTENCRYPTION_OBJ_P(getThis())->client_encryption, &keyid, keyaltname, &key_doc, &error)) {
+		phongo_throw_exception_from_bson_error_t(&error);
+		goto cleanup;
+	}
+
+	RETVAL_OPTIONAL_BSON_T(key_doc);
+
+cleanup:
+	bson_value_destroy(&keyid);
+	bson_destroy(&key_doc);
+} /* }}} */
+
+/* {{{ proto object MongoDB\Driver\ClientEncryption::rewrapManyDataKey(array|object $filter[, array $options = array()])
+   Decrypts multiple data keys and (re-)encrypts them with a new masterKey, or
+   with their current masterKey if a new one is not given. Returns an object
+   corresponding to the internal libmongoc result. */
+static PHP_METHOD(MongoDB_Driver_ClientEncryption, rewrapManyDataKey)
+{
+	zval*        zfilter       = NULL;
+	zval*        options       = NULL;
+	bson_t       filter        = BSON_INITIALIZER;
+	char*        provider      = NULL;
+	zend_bool    free_provider = false;
+	bson_t*      masterkey     = NULL;
+	bson_error_t error         = { 0 };
+	bson_t       reply         = BSON_INITIALIZER;
+
+	mongoc_client_encryption_rewrap_many_datakey_result_t* result = NULL;
+	const bson_t*                                          bulk_write_result;
+
+	PHONGO_PARSE_PARAMETERS_START(1, 2)
+	PHONGO_PARAM_ARRAY_OR_OBJECT(zfilter)
+	Z_PARAM_OPTIONAL
+	Z_PARAM_ARRAY(options)
```

Comment:
While this is technically correct, we should change this to also accept null for consistency with older APIs.

Related to https://github.com/mongodb/mongo-php-driver/pull/1346#issuecomment-1216712630